src/events/componentEvents/suggestion-accept.ts

```diff
@@ -0,0 +1,35 @@
+import { ComponentEvent } from "../../utils/ComponentEvent";
+import { Logger } from "../../utils/Logger";
+import { config } from "../../config";
+import type { GuildTextBasedChannel } from "discord.js";
+import type { APIEmbed } from "discord-api-types/payloads";
+
+export const event = new ComponentEvent(
+	"suggestion-accept",
+	async (interaction) => {
+		try {
+			if (!interaction.isButton() || !interaction.inCachedGuild()) return;
+
+			await interaction.deferUpdate();
```

Comment:
This is bad. You defer a button without replying afterwards and instead, you delete the corresponding message. I'm not sure why you aren't replying to the interaction instead of sending a message to the channel.